km_api/know_me/filters.py

```diff
@@ -32,6 +32,35 @@ def filter_list_queryset(self, request, queryset, view):
         return queryset.filter(user=request.user)
 
 
+class ListEntryFilterBackend(DRYPermissionFiltersBase):
+    """
+    Filter for listing ``ListEntry`` instances.
+    """
+
+    def filter_list_queryset(self, request, queryset, view):
+        """
+        Filter list entries for a ``list`` action.
+
+        Args:
+            request:
+                The request being made:
+            queryset:
+                A queryset containing the objects to filter.
+            view:
+                The view being accessed.
+
+        Returns:
+            A queryset containing the list entries accessible to the
+            user making the request.
+        """
+        item = get_object_or_404(
+            models.ProfileItem,
+            pk=view.kwargs.get('pk'),
+            profile__km_user__user=request.user)
```

Comment:
This won't do what you want it to. Profile items don't have a parent `profile`, they have a parent `topic`. That means this line of the query should be `topic__profile__km_user__user=request.user`.

Also, this part isn't tested. If you look at the tests for the other filter backends, you will see they test three things. It should only return the entries attached to that specific profile item, it should raise an error if there is no item with the provided ID, and it should raise an error if the requesting user doesn't have access.